tests/test_settings.py

```diff
@@ -1,40 +1,82 @@
+import sys
 from copy import deepcopy
 
 from django.core.exceptions import ImproperlyConfigured
-from django.test import TestCase
+from django.test import SimpleTestCase, override_settings
 from mock import patch
 from django_auth_adfs.config import django_settings
-from django_auth_adfs.config import Settings
 
+from .custom_config import Settings as CustomSettings
+
+
+class ReloadModuleMixin(object):
```

Comment:
This was a mess - because of the sentinel `settings` in `config.py`, we need to force re-import to bust the cache in `sys.modules`.

Maybe a `LazySettings` strategy should be followed so we can exclusively make use of `@override_settings`, but that seemed like to big of a refactor, at the cost of some... annoying testing code.

Reply:
Don't bother about the test too much. I'm busy reviewing the test suite in general as it's not how I would like it to be at the moment. Way too many patching stuff to get some of the settings changed.